src/NuGet.Clients/NuGet.Console/OutputConsole/BaseOutputConsole.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.VisualStudio;
+
+namespace NuGetConsole
+{
+    /// <summary>
+    /// Extends <see cref="SharedOutputConsole"/> and adds the shared implementation for <see cref="IConsoleDispatcher"/> and <see cref="IConsole"/>.
+    /// This class declares the methods that need implemented as abstract.
+    /// It allows to avoid diferences in experience across VS Online Environments and VS.
+    /// </summary>
+    internal abstract class BaseOutputConsole : SharedOutputConsole, IConsoleDispatcher, IConsole
+    {
+        public abstract void StartConsoleDispatcher();
+
+        public void Start()
```

Comment:
Does the Start needs to be thread safe?

Reply:
Not really. In theory, it'd be nice, but the way it's implemented we don't need to bother with it.
I just did some refactoring, so the way Start is implemented remains the same.